src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583MessageConnection.java

```diff
@@ -0,0 +1,145 @@
+                        , Integer.parseInt(threadSafeTime), TimeUnit.SECONDS, workQueue);
+            } else {
+                BlockingQueue<Runnable> workQueue = new LinkedBlockingQueue<Runnable>(
+                        Integer.parseInt(ISO8583Constant.THREAD_QLEN));
+                threadPool = new ThreadPoolExecutor(Integer.parseInt(ISO8583Constant.CORE_THREADS),
+                        Integer.parseInt(ISO8583Constant.MAX_THREADS), Integer.parseInt(ISO8583Constant.KEEP_ALIVE),
+                        TimeUnit.SECONDS, workQueue);
+            }
+        } catch (NumberFormatException e) {
+            handleException("One of the property or properties of thread specified is of an invalid type", e);
+        }
+        return threadPool;
+    }
+
+    /**
+     * create the server socket which is to accept a connection from a client.
+     */
+    public void run() {
+        try {
+            server = new ServerSocket(port);
+            log.info("Server is listening on port :" + port);
+            while (!listening) {
+                try {
+                    Socket socketConnection = server.accept();
+                    if (log.isDebugEnabled()) {
+                        log.debug("Client connected to socket: " + socketConnection.toString());
+                    }
+                    handleClientRequest(socketConnection, params);
+                } catch (IOException e1) {
+                    log.warn("Exception occurred while accept the connections", e1);
+                }
+            }
+        } catch (IOException e) {
+            handleException("Server could not listen on port " + port, e);
+        } finally {
+            try {
+                server.close();
+            } catch (IOException e) {
+                log.error("Error while closing the serverSocket", e);
+            }
+        }
+    }
+
+    /**
+     * destroy the inbound and connections.
+     */
+    public void destroyConnection() {
+        log.info("destroy the connection");
+        try {
+            if (!server.isClosed()) {
+                server.close();
+                log.info("Server stop the listening on port:" + port);
+                listening = false;
+            }
+        } catch (IOException e) {
+            handleException("Couldn't close the server", e);
+        }
+    }
+
+    /**
+     * handle the client request
+     *
+     * @param connection Socket connection
+     * @param params     the InboundProcessorParams
+     */
+    private void handleClientRequest(Socket connection, InboundProcessorParams params) {
+        try {
+            threadPool.execute(new ConnectionRequestHandler(connection, params));
+        } catch (RejectedExecutionException re) {
+            // If the pool is full complete the execution with the same thread
+            log.warn("Worker pool has reached the maximum capacity.");
```

Comment:
Just to warn the user that queue is full and  cannot add any more threads in the thread pool.
I have added the stack trace as well, please find that in following commit
7bdcc0493fa7623ea576d32be2c8b292f098467e